src/spawn.rs

```diff
@@ -1,166 +1,63 @@
-use std::future::Future;
-
-#[cfg(feature = "async_std")]
-pub use async_std_impl::*;
-#[cfg(feature = "smol")]
-pub use smol_impl::*;
+/// Spawns the given actor into the tokio runtime, returning an [`Address`](crate::Address) to it.
+///
+/// The 2nd parameter is the mailbox size with `None` creating an unbounded mailbox.
 #[cfg(feature = "tokio")]
-pub use tokio_impl::*;
-#[cfg(feature = "wasm_bindgen")]
-pub use wasm_bindgen_impl::*;
+#[cfg_attr(docsrs, doc(cfg(feature = "tokio")))]
+pub fn spawn_tokio<A>(actor: A, mailbox_size: Option<usize>) -> crate::Address<A>
+where
+    A: crate::Actor<Stop = ()>,
+{
+    let (address, context) = crate::Context::new(mailbox_size);
+
+    tokio::spawn(context.run(actor));
 
-/// An `Spawner` represents anything that can spawn a future to be run in the background. This is
-/// used to spawn actors.
-pub trait Spawner {
```

Comment:
How would you feel about an alternate API like
```rust
fn spawn<A: Actor, R: Spawner>(act: A, rt: R) -> Address<A> {
    
    let (address, context) = crate::Context::new(mailbox_size);
    rt.spawn(context.run(actor));
    address
}
```
`Spawner` could be implemented for `tokio::runtime::Handle`, and also for a variety of unit structs such as `TokioGlobal`, `SmolGlobal`, `WasmLocal`, etc, and it could be called like `xtra::spawn(act, xtra::rt::TokioGlobal)`. The advantage here would be that we could cut down on the number of functions in the public API, and it might simplify things from an understanding perspective - just swap out the `rt` arg to change where it's spawned to

Reply:
I am not sure the indirection is worth it? Typically, a project will only use one kind of runtime and having to specify the runtime as an argument is always another type you need to type / import.

In reality, this argument is always going to be a constant because of the same reason so it feels like expressing it via functions is better than arguments (which are typically only used for things that can be dynamic).

I've also mentioned it in the PR description: These functions are an 80/20 solution to me: Serve 80% of usecases with 20% of the effort. The implementation is so simple that a project with different needs can write one in 2 minutes. In fact, I'd expect most projects to eventually have their own `spawn_actor` function that standardizes the mailbox size and runtime for the project.

Reply:
If it wouldn't be such an ergonomic hit, I'd actually love to remove any runtime integration so we are truly runtime agnostic. Unfortunately, I don't think that is possible if we want to provide a way of spawning an actor in a single line of code.

Reply:
> If it wouldn't be such an ergonomic hit, I'd actually love to remove any runtime integration so we are truly runtime agnostic. Unfortunately, I don't think that is possible if we want to provide a way of spawning an actor in a single line of code.

Yea, AFAIK there is currently no way to abstract over all runtimes unless you do it yourself.

Reply:
> The advantage here would be that we could cut down on the number of functions in the public API, and it might simplify things from an understanding perspective - just swap out the `rt` arg to change where it's spawned to

Whilst we cut down the number of functions in the public API, the number of public API elements is actually larger:

- `Spawner`
- `spawn`
- 1 struct per runtime, i.e. 4

That is 6 elements vs 4 in the proposed solution, plus, `spawn` has 1 more parameter and it has here.

I am not sure if number of API elements is a good metric here but I think the increased number of moving parts makes it a bit harder to understand.

Reply:
We could provide both, release an RC and ask users, which one they prefer?

Reply:
That makes sense, I think. Happy to merge this for now though and we can maybe add the other API in a followup?

Reply:
Cool! It is ready from my end :)